### PR TITLE
Issue3 log in backend

### PR DESCRIPTION
**Pull request by roosavihlman**
_Task: https://github.com/APUOPE-RE/backend/issues/3_

Created to backend:
- Log in controller
- One end point, which fetches from db if there exists user with given email and passqord
- Service class
- User and UserCredentials classes
- Unit tests for successful and unsuccessful login attempt

_Note_ these are just initial classes and services done. These can be developed when implementation goes forward -> just to get the big picture up and running._

Testing can be done e.g. with user:
```
INSERT INTO apuope.users (username, email, password_hash)
VALUES ('someone', 'someone@gmail.com', '123456789');

```
Frontend PR: https://github.com/APUOPE-RE/frontend/pull/10

### CODE REVIEW

File: src/test/java/com/apuope/apuope_re/services/LogInServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+class LogInServiceTest {
+    @Mock
+    private UserRepository userRepository;  // Mock the repository
+
+    @InjectMocks
+    private LogInService logInService;  // Inject mock repository into the service
```
Comment (williamirva) [inline src/test/java/com/apuope/apuope_re/services/LogInServiceTest.java:23]: These comments can be deleted in my opinion

File: src/main/java/com/apuope/apuope_re/services/LogInService.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import com.apuope.apuope_re.entities.User;
+import com.apuope.apuope_re.repositories.UserRepository;
+import com.apuope.apuope_re.dto.UserCredentials;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+
+@Service
+public class LogInService {
+    @Autowired
```
Comment (williamirva) [inline src/main/java/com/apuope/apuope_re/services/LogInService.java:12]: I would delete auto wired and create a class constructor instead:

```
@Service
public class LogInService {
    private final UserRepository userRepository;

    public LogInService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

```

File: src/test/java/com/apuope/apuope_re/services/LogInServiceTest.java (hunk 1 above)
Comment (roosavihlman) [reply src/test/java/com/apuope/apuope_re/services/LogInServiceTest.java]: Fixed

File: src/main/java/com/apuope/apuope_re/services/LogInService.java (hunk 2 above)
Comment (roosavihlman) [reply src/main/java/com/apuope/apuope_re/services/LogInService.java]: Fixed

**Review by williamirva — APPROVED** [review]

**Pull request merged**